server/requestlog/requestlog.go

```diff
@@ -114,6 +115,7 @@ type Entry struct {
 	Latency            time.Duration
 	TraceID            trace.TraceID
 	SpanID             trace.SpanID
+	Request            *http.Request
```

Comment:
@zombiezen thoughts?

This makes me nervous because Request is potentially mutable by the caller, all of the other info in this struct is immutable.

If it's just URL info you want, we could consider cloning the net.URL, but it's possible that the easiest way to do that is to re-parse the URL string (e.g., that's the first result for https://www.google.com/search?q=golang+clone+url for me anyway), which you can already do.

Reply:
I'm in agreement with @vangent here: the current struct is simple and immutable now. The `*http.Request` struct is large and contains many fields not relevant to an `Entry`, so I don't think it's appropriate to add one.

I could be convinced that this struct should include the HTTP headers, but that's beyond the use case outlined in #2734 which only needed the URL.

Reply:
Totally fair. Would you be game if I just add the Header and Body? I can live without a clone of net.URL and will just parse the string back myself.

What I'm trying to get at is - as part of an API error response, I'm adding the opencensus TraceID that you've so nicely packaged into Entry. For customer support issues, I'd like to be able to find that TraceID in logs and inspect the request headers and body for debugging. It can be helpful. If this is outside the scope of what you've intended for Entry, I can understand that and will create a a separate log to get these details. I know the Body being an `io.Closer` makes it tricky, but I'd just leave it as is and separately dump it similar to how it's done in `httputil.DumpRequest`.

Thank you both again for your time and consideration!

Dan

Reply:
Added a 2nd commit with these changes, if it helps!